dashboard/lib/policies/child_account.rb

```diff
@@ -76,10 +73,24 @@ def self.show_cap_state_modal?(user)
   # Checks if a user is affected by a state policy but was created prior to the
   # policy going into effect.
   def self.user_predates_policy?(user)
-    parent_permission_required?(user) && (
-      user.created_at < STATE_POLICY[user.us_state][:start_date] ||
-      user.authentication_options.any?(&:google?)
-    )
+    return false unless parent_permission_required?(user)
+    return false unless state_policy(user)
+    policy_start_date = state_policy(user)[:start_date]
+
+    user.created_at < policy_start_date ||
+      user.authentication_options.any?(&:google?) ||
+      # [P20-937] CPA students created after the policy came into effect but
+      # before '2024/05/18' might have had their us_state field not managed
+      # properly, so we will allow them to be treated the same as users who
+      # predate CPA starting in 2023/07/1.
+      (
+        user.created_at < Date.parse('2024-05-18T00:00:00Z') &&
+        user.authentication_options.any? do |ao|
```

Comment:
Are we getting much value from checking these remaining auth option types? We're already handling Google on line 81, and LTI, Clever, and word/picture are all exempted elsewhere, right? So by including email, Microsoft, and Facebook, we're pretty much covering everything. Could we just do `user.created_at < Date.parse('2024-05-18T00:00:00Z')` to keep it simple?

Reply:
The rational was that I was trying to closely match the fix with the bug report, but you are right, this could be simplified to just checking the created date.

Reply:
oh interesting. yeah I suppose there is no test for a provider that would trigger this to be something different than Facebook, etc... because there isn't any situation?